leakcanary-android/src/main/java/com/squareup/leakcanary/LeakCanary.java

```diff
@@ -142,7 +144,11 @@ public static String leakInfo(Context context, HeapDump heapDump, AnalysisResult
    * a different process than the normal app process.
    */
   public static boolean isInAnalyzerProcess(Context context) {
-    return isInServiceProcess(context, HeapAnalyzerService.class);
+    // This only needs to be computed once per process.
+    if (isInAnalyzerProcess == null) {
```

Comment:
If you read to a local you avoid having to double-read the volatile. You have to update the local and static inside the `if` then.